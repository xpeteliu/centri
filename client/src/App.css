.mainContent {
  margin: 20px;
}

body {
  background-color: #E4F0D1;
}

.btn-primary, .btn-primary:hover, .btn-primary:active, .btn-primary:visited {
  background-color: #96BB88 !important;
  border-color: #96BB88 !important;
  color: #000000 !important;
}

.btn-secondary, .btn-secondary:hover, .btn-secondary:active, .btn-secondary:visited {
  background-color: #C1D9B8 !important;
  border-color: #C1D9B8 !important;
  color: #000000 !important;
}

.headerBar {
  background-color: #C1D9B8 !important;
}

.groupListItem {
  background-color: #FFFFFF;
  height: 80px;
  padding: 20px;
}

/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-float infinite 3s ease-in-out;
  }
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: rgb(112, 76, 182);
}

@keyframes App-logo-float {
  0% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(10px);
  }
  100% {
    transform: translateY(0px);
  }
} */
