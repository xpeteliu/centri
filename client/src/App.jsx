import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import HomePage from './components/HomePage/HomePage';
import LoginPage from './components/LoginPage/LoginPage';
import './App.css';

const Home = () => (
  <HomePage />
);

const Login = () => (
  <LoginPage />
);

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/" component={Home} />
          <Route path="/login" component={Login} />
        </div>
      </Router>
    );
  }
}

export default App;
