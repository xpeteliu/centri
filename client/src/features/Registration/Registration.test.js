import React from 'react';
import {
  act, render, screen,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { setupServer } from 'msw/node';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import App from '../../App';

describe('Registration page', () => {
  const mockServer = setupServer(
    rest.post('/user/',
      (req, res, ctx) => (req.body.username === 'duplicateName'
        ? res(ctx.status(409), ctx.json({ message: 'The username has been registered' }))
        : res(ctx.status(204)))),
  );

  const history = createMemoryHistory({ initialEntries: ['/register'] });

  beforeAll(() => { mockServer.listen(); });

  afterAll(() => { mockServer.close(); });

  beforeEach(() => {
    act(() => {
      render(
        <Router history={history}>
          <App />
        </Router>,
      );
    });
  });

  test('should properly handle valid registration info', async () => {
    const inputBoxes = await screen.findAllByRole('textbox');
    const submitBtn = await screen.findByRole('button', { name: /sign up/i });
    expect(inputBoxes.length).toEqual(4);
    act(() => {
      userEvent.type(inputBoxes[0], 'testName');
      userEvent.type(inputBoxes[1], 'testName@email.com');
      userEvent.type(inputBoxes[2], 'testPwd');
      userEvent.type(inputBoxes[3], 'testPwd');
      userEvent.click(submitBtn);
    });
    expect(history.location.pathname).toEqual('/login');
  });

  test('should properly handle occupied username', async () => {
    const inputBoxes = await screen.findAllByRole('textbox');
    const submitBtn = await screen.findByRole('button', { name: /sign up/i });
    expect(inputBoxes.length).toEqual(4);
    act(() => {
      userEvent.type(inputBoxes[0], 'duplicateName');
      userEvent.type(inputBoxes[1], 'duplicateName@email.com');
      userEvent.type(inputBoxes[2], 'testPwd');
      userEvent.type(inputBoxes[3], 'testPwd');
      userEvent.click(submitBtn);
    });
    expect(history.location.pathname).toEqual('/register');
  });

  test('should properly handle different passwords for the 2 input boxes', async () => {
    const inputBoxes = await screen.findAllByRole('textbox');
    const submitBtn = await screen.findByRole('button', { name: /sign up/i });
    expect(inputBoxes.length).toEqual(4);
    act(() => {
      userEvent.type(inputBoxes[0], 'testName');
      userEvent.type(inputBoxes[1], 'testName@email.com');
      userEvent.type(inputBoxes[2], 'testPwd');
      userEvent.type(inputBoxes[3], 'diffPwd');
      userEvent.click(submitBtn);
    });
    expect(history.location.pathname).toEqual('/login');
  });
});
