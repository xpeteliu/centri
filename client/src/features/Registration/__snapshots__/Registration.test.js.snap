// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="h-100 container"
    >
      <div
        class="h-100 row"
      >
        <div
          class="my-auto col"
        >
          <div
            class="mx-auto card"
            style="width: 30rem;"
          >
            <div
              class="card-header"
            >
              <h2>
                Sign Up
              </h2>
            </div>
            <div
              class="card-body"
            >
              <form
                class="container"
                id="formLogin"
                style="width: 25rem;"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="ms-0 form-label"
                        for="inputRegistrationUsername"
                      >
                        Username
                      </label>
                      <input
                        class="form-control"
                        id="inputRegistrationUsername"
                        placeholder="Enter username here"
                        required=""
                        type="text"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="form-label"
                        for="inputRegistrationEmail"
                      >
                        Email
                      </label>
                      <input
                        class="form-control"
                        id="inputRegistrationEmail"
                        placeholder="Enter email here"
                        required=""
                        type="email"
                      />
                      <small
                        class="text-muted form-text"
                      >
                        Your email will be kept confidential with us.
                      </small>
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="form-label"
                        for="inputRegistrationPassword0"
                      >
                        Password
                      </label>
                      <input
                        class="form-control"
                        id="inputRegistrationPassword0"
                        pattern="^(?=.*[A-Za-z])(?=.*\\\\d)[A-Za-z\\\\d]{8,}$"
                        placeholder="Enter password here"
                        required=""
                        title="A password should contain >=8 characters and include both letters and numbers"
                        type="password"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 text-start"
                    >
                      <label
                        class="form-label"
                        for="inputRegistrationPassword1"
                      >
                        Verify Password
                      </label>
                      <input
                        class="form-control"
                        id="inputRegistrationPassword1"
                        pattern="^(?=.*[A-Za-z])(?=.*\\\\d)[A-Za-z\\\\d]{8,}$"
                        placeholder="Enter the same password as above"
                        required=""
                        title="A password should contain >=8 characters and include both letters and numbers"
                        type="password"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                  style="margin-top: 1rem;"
                >
                  <div
                    class="col"
                  >
                    <button
                      class="btn btn-primary"
                      form="formLogin"
                      type="submit"
                    >
                      Register
                    </button>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="container card-footer"
            >
              <div
                class="text-center row"
              >
                <div
                  class="col"
                >
                  <a
                    class="card-link"
                    href="/login"
                  >
                    Already have an account?
                  </a>
                </div>
                <div
                  class="col"
                >
                  <a
                    class="card-link"
                    href="/"
                  >
                    Back to Home Page
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
